module1/homework/make_unique/make_unique.hpp

```diff
@@ -1,11 +1,19 @@
 #pragma once
 
 #include <memory>
+#include <type_traits>
 
 namespace cs
 {
     template<typename ObjectType, typename... ArgTypes>
-    std::unique_ptr<ObjectType> make_unique(ArgTypes&&... args) {
+    std::enable_if_t<!std::is_array_v<ObjectType>, std::unique_ptr<ObjectType>>
+    make_unique(ArgTypes&&... args) {
         return std::unique_ptr<ObjectType>(new ObjectType{std::forward<ArgTypes>(args)...});
     }
+
+    template<typename ArrayType>
+    std::enable_if_t<std::is_array_v<ArrayType>, std::unique_ptr<ArrayType>>
```

Comment:
Good. There is also a bit simpler SFINAE with enable_if as additional template parameter instead of a return type.